lib/rules/no-constant-condition.js

```diff
@@ -115,15 +118,74 @@ module.exports = {
         }
 
         /**
-         * Reports when the given node contains a constant condition.
+         * Tracks when the given node contains a constant condition.
          * @param {ASTNode} node The AST node to check.
          * @returns {void}
          * @private
          */
         function checkConstantCondition(node) {
             if (node.test && isConstant(node.test, true)) {
+                setOfConstantLoops.add(node);
+            }
+        }
+
+        /**
+         * Reports when the set contains the given constant condition node
+         * @param {ASTNode} node The AST node to check.
+         * @returns {void}
+         * @private
+         */
+        function checkConstantConditionLoopInSet(node) {
+            if (setOfConstantLoops.has(node)) {
+                setOfConstantLoops.delete(node);
+                context.report({ node, message: "Unexpected constant condition." });
+            }
+        }
+
+        /**
+         * Reports when the given node contains a constant condition.
+         * @param {ASTNode} node The AST node to check.
+         * @returns {void}
+         * @private
+         */
+        function checkIfConstant(node) {
+            if (node.test && isConstant(node.test, true)) {
+                context.report({ node, message: "Unexpected constant condition." });
+            }
+        }
+
+        /**
+         * Clears setOfConstantLoops
+         * @returns {void}
+         * @private
+         */
+        function clearSetOfLoops() {
+            setOfConstantLoops.clear();
+        }
+
+        /**
+         * Stores current set of constant loops in loopSetStack temporarily
+         * and uses a new set to track constant loops
+         * @returns {void}
+         * @private
+         */
+        function enterFunction() {
+            loopSetStack.push(setOfConstantLoops);
+            setOfConstantLoops = new Set();
+        }
+
+        /**
+         * Reports when the set still contains stored constant conditions
+         * @param {ASTNode} node The AST node to check.
+         * @returns {void}
+         * @private
+         */
+        function exitFunction(node) {
+            if (setOfConstantLoops.size > 0) {
```

Comment:
Will this condition ever occur? It seems like any loop in the set would have already been removed when the loop was exited, so it will never still be in the set when the function is exited.